draft-ietf-quic-transport.md

```diff
@@ -2611,173 +2612,74 @@ CONNECTION_CLOSE or RST_STREAM frame. Error codes share a common code space.
-
-QUIC_STREAM_DATA_AFTER_TERMINATION (0x80000002):
-: There were data frames after the a fin or reset.
-
-QUIC_INVALID_PACKET_HEADER (0x80000003):
-: Control frame is malformed.
-
-QUIC_INVALID_FRAME_DATA (0x80000004):
-: Frame data is malformed.
-
-QUIC_MULTIPLE_TERMINATION_OFFSETS (0x80000005):
-: Multiple final offset values were received on the same stream
-
-QUIC_STREAM_CANCELLED (0x80000006):
-: The stream was cancelled
-
-QUIC_CLOSED_CRITICAL_STREAM (0x80000007):
-: A stream that is critical to the protocol was closed.
-
-QUIC_MISSING_PAYLOAD (0x80000030):
-: The packet contained no payload.
-
-QUIC_INVALID_STREAM_DATA (0x8000002E):
-: STREAM frame data is malformed.
-
-QUIC_UNENCRYPTED_STREAM_DATA (0x8000003D):
-: Received STREAM frame data is not encrypted.
-
-QUIC_MAYBE_CORRUPTED_MEMORY (0x80000059):
-: Received a frame which is likely the result of memory corruption.
-
-QUIC_INVALID_RST_STREAM_DATA (0x80000006):
-: RST_STREAM frame data is malformed.
-
-QUIC_INVALID_CONNECTION_CLOSE_DATA (0x80000007):
-: CONNECTION_CLOSE frame data is malformed.
-
-QUIC_INVALID_GOAWAY_DATA (0x80000008):
-: GOAWAY frame data is malformed.
-
-QUIC_INVALID_WINDOW_UPDATE_DATA (0x80000039):
-: WINDOW_UPDATE frame data is malformed.
-
-QUIC_INVALID_BLOCKED_DATA (0x8000003A):
-: BLOCKED frame data is malformed.
-
-QUIC_INVALID_PATH_CLOSE_DATA (0x8000004E):
-: PATH_CLOSE frame data is malformed.
-
-QUIC_INVALID_ACK_DATA (0x80000009):
-: ACK frame data is malformed.
-
-QUIC_INVALID_VERSION_NEGOTIATION_PACKET (0x8000000A):
-: Version negotiation packet is malformed.
-
-QUIC_INVALID_PUBLIC_RST_PACKET (0x8000000b):
-: Public RST packet is malformed.
-
-QUIC_DECRYPTION_FAILURE (0x8000000c):
-: There was an error decrypting.
-
-QUIC_ENCRYPTION_FAILURE (0x8000000d):
-: There was an error encrypting.
-
-QUIC_PACKET_TOO_LARGE (0x8000000e):
-: The packet exceeded kMaxPacketSize.
-
-QUIC_PEER_GOING_AWAY (0x80000010):
-: The peer is going away. May be a client or server.
-
-QUIC_INVALID_STREAM_ID (0x80000011):
-: A stream ID was invalid.
-
-QUIC_INVALID_PRIORITY (0x80000031):
-: A priority was invalid.
-
-QUIC_TOO_MANY_OPEN_STREAMS (0x80000012):
-: Too many streams already open.
-
-QUIC_TOO_MANY_AVAILABLE_STREAMS (0x8000004c):
-: The peer created too many available streams.
-
-QUIC_PUBLIC_RESET (0x80000013):
-: Received public reset for this connection.
-
-QUIC_INVALID_VERSION (0x80000014):
-: Invalid protocol version.
-
-QUIC_INVALID_HEADER_ID (0x80000016):
-: The Header ID for a stream was too far from the previous.
-
-QUIC_INVALID_NEGOTIATED_VALUE (0x80000017):
-: Negotiable parameter received during handshake had invalid value.
+INTERNAL_ERROR (0x80000001):
 
-QUIC_DECOMPRESSION_FAILURE (0x80000018):
-: There was an error decompressing data.
+: The endpoint encountered an internal error and cannot continue with the
+  connection.
 
-QUIC_NETWORK_IDLE_TIMEOUT (0x80000019):
-: The connection timed out due to no network activity.
+ENHANCE_YOUR_CALM (0x80000002):
```

Comment:
Where's this intended to be used? It's not clear to me what exactly the endpoint wants the peer to reduce generation of -- Data? Acks? Window updates? -- and what the peer is expected to do when this error code is received. Is this a stream or connection error? It would be an odd time to say reduce generation  of something if used as a connection error... the connection is now closed. Does this error code mean that the peer SHOULD NOT reconnect?